docs/migration-guides/4.0.md

```diff
@@ -0,0 +1,81 @@
+If you haven't written a custom ruleset, never used a config and use Spectral via CLI only,
+it's very likely you are unaffected by any of the breaking changes we introduced,
+therefore you might upgrade your Spectral version right away.
+Our docs have been updated, so you can always refer to them, but to make the transition less painful,
+this migration guide covers the most notable changes.
+
+Note, 3.0 version was a bit of a transition-version, without any significant changes, besides the config file we removed in 4.0.
+That said, the following migration is kind of 2.x/3.x to 4.x.
+
+
+### I have own rulesets
+
+1. summary property is gone
+
+We introduced a more powerful `message` property. It not only lets you define a static message, but also lets you compose them.
+If you want to have error from function printed, you can do so as follows
+```js
+message: '{{error}}'
+```
+Printing description is equally straightforward:
+```js
+message: '{{description}}'
+```
+If you want a property key to be printed instead...
+```js
+message: '{{property}}'
+```
+Bear in mind, property might be an empty string.
+You can also compose a fancier message if you'd like to.
+```js
+message: 'Errored at {{property}}... Error message is: {{error}}'
+```
+
+Let us know if there is any shorthand you'd like to see.
+
+A real-world example of the migration can be found here https://github.com/stoplightio/spectral/issues/367
+ 
+
+2. jsonpath is swapped with jsonpath-plus
+
+There shouldn't be any significant differences, but jsonpath-plus does come with its own syntax for special selectors.
+Please refer to their docs for the comprehensive list. https://github.com/s3u/JSONPath#features
+You can also take a look at changes we had to make internally https://github.com/stoplightio/spectral/pull/347/commits/c79691da636f76c5b0ab5c611779d6be16d32292
+
+### I use Spectral via CLI
+
+1. Config is gone...
+
+...here is no way to restore it.
```

Comment:
this could be "The config files (which was `spectral.yml` by default, or using the  `--config` was a combination of things that could all be done as CLI flags, and was just meant to reduce how much you needed to type those things out. The important stuff has been replaced with looking for a drefault ruleset, called `.spectral.yml` (notice the `.` prefix) and loading a ruleset and skipping a rule would now look like this: (before and after)

Reply:
@philsturgeon done.
Besides that, added a note on `recommended` property.